src/Spice86.Core/Emulator/OperatingSystem/DosFilePathResolver.cs

```diff
@@ -0,0 +1,172 @@
+namespace Spice86.Core.Emulator.OperatingSystem;
+
+using Spice86.Shared.Emulator.Errors;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+/// <inheritdoc />
+public class DosFilePathResolver : IDosFilePathResolver {
+    private readonly ILoggerService _loggerService;
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    /// <param name="loggerService">The logger service implementation.</param>
+    public DosFilePathResolver(ILoggerService loggerService) => _loggerService = loggerService;
+    
+    /// <inheritdoc />
+    public string GetHostParentDirectory(string path) => Directory.GetParent(path)?.FullName ?? path;
```

Comment:
Not sure this has something to do with Host. Maybe just "GetParentDirectory"? Also, could it be used in places where Directory.GetParent is used?

Reply:
Renamed, and used everywhere it made sense.